java-frontend/src/main/java/org/sonar/java/se/checks/CustomUnclosedResourcesCheck.java

```diff
@@ -45,13 +45,13 @@
 @RuleTemplate
 public class CustomUnclosedResourcesCheck extends SECheck {
 
-  class Status {
-    Object OPENED = new Object();
-    Object CLOSED = new Object();
-  }
+  static class ResourceStatus implements ObjectConstraint.Status {
 
-  private final Status status = new Status();
+  }
 
+  //these fields can not be static, because we need different status instance for every instantiation of this rule (template rule) see SONARJAVA-1624
```

Comment:
Let's try to keep it short :
 //see SONARJAVA-1624 fields cannot be static: different instance are needed for every instance of this template rule